purchase_order_cancel/wizard/purchase_order_cancel.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, _
+
+
+class PurchaseOrderCancel(models.TransientModel):
+    _name = 'purchase.order.cancel'
+
+    @api.model
+    def _cancel_purchase_order(self, orders):
+        return orders.button_cancel()
+
+    @api.multi
+    def action_cancel_purchase_order(self):
+        active_ids = self._context.get('active_ids', [])
+        active_model = self._context.get('active_model', 'purchase.order')
+        orders = self.env[active_model].browse(active_ids)
+        self._cancel_purchase_order(orders)
```

Comment:
@mra-odoo We removed the checking here as the `button_cancel` method should still throw error if the purchase order cannot be cancelled. Please let me know if there is any reason(s) of performing the checking here.

Reply:
@TimLai125 Yes you are correct. Odoo standard will  handle that. Thanks.